app/jobs/create_derivatives_job_decorator.rb

```diff
@@ -27,6 +32,14 @@ def perform(file_set, file_id, filepath = nil)
       file_set.update_index
       file_set.parent.update_index if parent_needs_reindex?(file_set)
     end
+
+    private
+
+      def reschedule(file_set, file_id, filepath = nil)
```

Comment:
not that it needs to hold up this pr, but should we consider putting a limit on the amount of times this reschedule will happen?

Reply:
totally, commit coming soon.